### DISCUSSION TITLE
UCCSD with complex ERI integrals?

### DISCUSSION THREAD

**Discussion by Physfock** [pyscf/pyscf#2628] [Q&A]
It appears that UCCSD is not functioning correctly with custom complex ERI, whereas RCCSD is operating effectively. How to resolve this issue?

Example:

mf = scf.UHF(mol)
smf.get_hcore = lambda *args: h_complex
mf.get_ovlp  = lambda *args: overlap_complex
mf._eri      = eri_complex
mf.kernel()

mycc = cc.UCCSD(mf).run()

I have in terminal: 

.local/lib/python3.10/site-packages/pyscf/cc/ccsd.py", line 489, in _contract_s4vvvv_t2
    assert (t2.dtype == numpy.double)
AssertionError

**Comment by sunqm** [comment]
Currently, the UCCSD code in PySCF does not support complex integrals directly. There are a few ways you could consider:
* If you're comfortable with some coding, you can modify the `pyscf.pbc.cc.kccsd_uhf` code. By removing the loop over k-points, you can adapt it to the UCCSD with complex integrals. 
* If you’re looking for a quicker solution, I'd recommend switching to the GCCSD module.
